crates/rover-client/src/shared/check_response.rs

```diff
@@ -1,159 +1,336 @@
+                ));
+                msg.push_str(operations_response.get_output().as_str());
+                msg.push('\n');
+            }
+        }
 
-impl SkipOperationsCheckResponse {
-    pub fn to_output(&self) -> String {
-        let mut msg = if self.core_schema_modified {
-            "There were no changes detected in the composed API schema, but the core schema was modified.".to_string()
-        } else {
-            "There were no changes detected in the composed schema.".to_string()
-        };
+        if let Some(lint_response) = &self.maybe_lint_response {
+            if !lint_response.diagnostics.is_empty() {
+                msg.push_str(&Self::task_title(
+                    "Lint Check",
+                    lint_response.task_status.clone(),
+                ));
+                msg.push_str(lint_response.get_output().as_str());
+                msg.push('\n');
+            }
+        }
 
-        if let Some(url) = &self.target_url {
-            msg.push_str("\n\n");
-            msg.push_str("View full details at: ");
-            msg.push_str(url);
-        };
+        if let Some(downstream_response) = &self.maybe_downstream_response {
+            if !downstream_response.blocking_variants.is_empty() {
+                msg.push_str(&Self::task_title(
+                    "Downstream Check",
+                    downstream_response.task_status.clone(),
+                ));
+                msg.push_str(downstream_response.get_output().as_str());
+                msg.push('\n');
+            }
+        }
 
         msg
     }
 
     pub fn get_json(&self) -> Value {
-        json!(self)
+        let mut tasks: Vec<Value> = Vec::new();
+
+        if let Some(operations_response) = &self.maybe_operations_response {
+            let mut operation_json = json!(operations_response);
+            operation_json
+                .as_object_mut()
+                .unwrap()
+                .insert("task_name".to_string(), json!("operation"));
+            tasks.push(operation_json);
+        }
+
+        if let Some(lint_response) = &self.maybe_lint_response {
+            let mut lint_json = json!(lint_response);
+            lint_json
+                .as_object_mut()
+                .unwrap()
+                .insert("task_name".to_string(), json!("lint"));
+            tasks.push(lint_json);
+        }
+
+        if let Some(downstream_response) = &self.maybe_downstream_response {
+            let mut downstream_json = json!(downstream_response);
+            downstream_json
+                .as_object_mut()
+                .unwrap()
+                .insert("task_name".to_string(), json!("downstream"));
+            tasks.push(downstream_json)
+        }
+
+        if let Some(core_schema_modified) = self.maybe_core_schema_modified {
+            json!({
+               "core_schema_modified": core_schema_modified,
+               "tasks": tasks
+            })
+        } else {
```

Comment:
This chunk of code feels a bit hacky to me - I think it would be great if we could keep calling `json!(self)` here to get the format we need instead

Reply:
This code was updated in https://github.com/apollographql/rover/pull/1637/commits/4d8f36e436f77479b8b6b211cc66bd0ba8d3af87 to return a map instead of an array. It sure help simplify it a bit but not nearly as nice as `json!(self)`.